console/ui/src/widgets/cluster-form/model/validation.ts

```diff
@@ -187,18 +187,33 @@ export const localFormSchema = (t: TFunction) =>
   });
 
 export const ClusterFormSchema = (t: TFunction) =>
-  yup
-    .object({
-      [CLUSTER_FORM_FIELD_NAMES.PROVIDER]: yup.object().required(),
-      [CLUSTER_FORM_FIELD_NAMES.ENVIRONMENT_ID]: yup.number(),
-      [CLUSTER_FORM_FIELD_NAMES.CLUSTER_NAME]: yup
-        .string()
-        .test('clusters should have proper naming', t('clusterShouldHaveProperNaming', { ns: 'validation' }), (value) =>
-          value.match(/^[a-z0-9][a-z0-9-]{0,23}$/g),
-        )
-        .required(t('requiredField', { ns: 'validation' })),
-      [CLUSTER_FORM_FIELD_NAMES.DESCRIPTION]: yup.string(),
-      [CLUSTER_FORM_FIELD_NAMES.POSTGRES_VERSION]: yup.number().required(t('requiredField', { ns: 'validation' })),
-    })
+  yup.object().shape({
+    [CLUSTER_FORM_FIELD_NAMES.PROVIDER]: yup.object().required(t('providerRequired')),
+    [CLUSTER_FORM_FIELD_NAMES.ENVIRONMENT_ID]: yup.number().required(t('environmentRequired')),
+    [CLUSTER_FORM_FIELD_NAMES.CLUSTER_NAME]: yup.string().required(t('clusterNameRequired')),
+    [CLUSTER_FORM_FIELD_NAMES.DESCRIPTION]: yup.string(),
+    [CLUSTER_FORM_FIELD_NAMES.POSTGRES_VERSION]: yup.number().required(t('postgresVersionRequired')),
+    [CLUSTER_FORM_FIELD_NAMES.EXISTING_CLUSTER]: yup.boolean().default(false),
+    [CLUSTER_FORM_FIELD_NAMES.DATABASE_SERVERS]: yup.array().when(CLUSTER_FORM_FIELD_NAMES.EXISTING_CLUSTER, {
+      is: true,
+      then: (schema) =>
+        schema
+          .of(
+            yup.object().shape({
+              [CLUSTER_FORM_FIELD_NAMES.HOSTNAME]: yup.string().required(t('hostnameRequired')),
+              [CLUSTER_FORM_FIELD_NAMES.IP_ADDRESS]: yup
+                .string()
+                .required(t('ipAddressRequired'))
+                .matches(
+                  /^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])$/,
+                  t('invalidIpAddress')
+                ),
+              [CLUSTER_FORM_FIELD_NAMES.LOCATION]: yup.string().required(t('locationRequired')),
```

Comment:
The location field is optional.
```suggestion
              [CLUSTER_FORM_FIELD_NAMES.LOCATION]: yup.string(),
```